sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/analysis/Analyzer.scala

```diff
@@ -607,14 +607,21 @@ class Analyzer(
     def apply(plan: LogicalPlan): LogicalPlan = plan resolveOperators {
       // Skip sort with aggregate. This will be handled in ResolveAggregateFunctions
       case sa @ Sort(_, _, child: Aggregate) => sa
```

Comment:
The above two issues have been covered in the rule `ResolveAggregateFunctions` when `Aggregate` is the child of `Sort`. Thus, the current PR keeps the existing logics without any change. 

This PR is to introduce a fix for aggregate functions when there exists the other operators between `Aggregate` and `Sort`.